Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -43,8 +44,9 @@ public FreezeOnMatchCustomization(Type targetType)
         /// that will be satisfied by the frozen specimen.
         /// </param>
         /// <exception cref="ArgumentNullException">
-        /// <paramref name="targetType"/> or<paramref name="matcher"/> is null.
+        /// <paramref name="targetType"/> or <paramref name="matcher"/> is null.
         /// </exception>
+        [Obsolete("Please use FreezeOnMatchCustomization(object, IRequestSpecification) instead.")]
```

Comment:
That may lead to the `InvalidCastException` if one passes not a `Type` instance:
```csharp
var sut = new FreezeOnMatchCustomization(new object());
// System.InvalidCastException: System.InvalidCastException : Unable to cast object
// of type 'System.Object' to type 'System.Type'.
```
Where the constructor is the following:
```csharp
    public FreezeOnMatchCustomization(object request)
            : this(request, new ExactTypeSpecification((Type)request))
```